javalib/src/main/scala/java/util/Collections.scala

```diff
@@ -0,0 +1,1131 @@
+        if (index < 0 || index >= n)
+          throw new IndexOutOfBoundsException
+        o
+      }
+    })
+  }
+
+  def reverseOrder[T](): Comparator[T] = {
+    new Comparator[T with Comparable[T]] {
+      def compare(o1: T with Comparable[T], o2: T with Comparable[T]): Int =
+        o2.compareTo(o1)
+    }.asInstanceOf[Comparator[T]]
+  }
+
+  def reverseOrder[T](cmp: Comparator[T]): Comparator[T] = {
+    class ReverseComparator extends Comparator[T] {
+      override def compare(o1: T, o2: T): Int = cmp.compare(o2, o1)
+    }
+    cmp match {
+      case _: io.Serializable => new ReverseComparator with io.Serializable
+      case _                  => new ReverseComparator
+    }
+  }
+
+  def enumeration[T](c: Collection[T]): Enumeration[T] = {
+    val it = c.iterator
+    new Enumeration[T] {
+      override def hasMoreElements: Boolean =
+        it.hasNext
+
+      override def nextElement(): T =
+        it.next()
+    }
+  }
+
+  def list[T](e: Enumeration[T]): ArrayList[T] = {
+    val arrayList = new ArrayList[T]
+    e.foreach(arrayList += _)
+    arrayList
+  }
+
+  def frequency(c: Collection[_], o: AnyRef): Int =
+    c.count(_ === o)
+
+  def disjoint(c1: Collection[_], c2: Collection[_]): Boolean = {
+    if (c1.size < c2.size)
+      !c1.exists(elem => c2.contains(elem))
+    else
+      !c2.exists(elem => c1.contains(elem))
+  }
+
+  def addAll[T](c: Collection[_ >: T], elements: Array[AnyRef]): Boolean =
+    c.addAll(asJavaCollection(elements.asInstanceOf[Array[T]]))
+
+  def newSetFromMap[E](map: Map[E, Boolean]): Set[E] = {
+    if (map.nonEmpty)
+      throw new IllegalArgumentException
+
+    new WrappedSet[E, Set[E]] {
+      override protected val inner: Set[E] =
+        map.keySet
+
+      override def add(e: E): Boolean =
+        map.put(e, true) === null
+
+      override def addAll(c: Collection[_ <: E]): Boolean =
+        c.foldLeft(false)((prev, elem) => map.put(elem, true) === null || prev)
+    }
+  }
+
+  @inline
+  private def modulo(a: Int, b: Int): Int = ((a % b) + b) % b
+
+  @inline
+  private def makeComparator[T <: jl.Comparable[T]]: Comparator[T] = {
+    new Comparator[T] {
+      final def compare(o1: T, o2: T): Int = o1.compareTo(o2)
+    }
+  }
+
+  @inline
+  private implicit def comparatorToOrdering[E](cmp: Comparator[E]): Ordering[E] = {
+    new Ordering[E] {
+      final def compare(x: E, y: E): Int = cmp.compare(x, y)
+    }
+  }
+
+  @inline
+  private def binarySearchCore[E](list: List[E], compareToKey: E => Int): Int = {
+    def notFound(insertionPoint: Int): Int = {
+      -insertionPoint - 1
+    }
+    list match {
+      case _: RandomAccess =>
+        @tailrec
+        def binarySearch(lo: Int, hi: Int): Int = {
+          if (lo < hi) {
+            val mid = lo + (hi - lo) / 2
+            val cmp = compareToKey(list(mid))
+            if (cmp === 0) mid
+            else if (cmp > 0) binarySearch(lo, mid)
+            else
+              binarySearch(mid + 1, hi)
```

Comment:
Put this on the same line as the `else`. (Or put all three branches on their dedicated line, but don't mix the two styles.)